docs/source/model_design/programming_vivarium/artifact_building/index.rst

```diff
@@ -29,12 +29,156 @@ Artifact Building
+is designed to take inputs from :code:`loader.py` and return the correct
+data key string.
+
+At the top of :code:`loader.py` you will see :code:`mapping`. The map provides
+inputs for :code:`data_keys.py` which returns the data key name. The second
+part of the line in :code:`mapping` is the function where the data key
+serves as an input. These functions are written below :code:`mapping`.
+
+Writing or Editing Data Keys
+----------------------------
+
+If you need to edit a data key, you can locate the corresponding function from
+:code:`mapping` at the top of :code:`loader.py` and simply edit the function
+to match your needs.
+
+It is important to note that some of the more common
+functions are stored in other repositories like :code:`vivarium_inputs`.
+Looking at these repositories can be helpful in understanding how the data is
+being pulled. However, generally RT should not edit functions in other
+engineering repositories. Talk to engineering if you think a function needs
+new functionality or try to add the needed updates in the :code:`loader.py` function instead.
+
+Another important call-out is that most of these functions are used for
+more than one data key. Check if you want to change how data is pulled for
+all data keys using that function, or only one, and edit accordingly.
+
+If you're being asked to add a new data key, or write an artifact from scratch,
+a good place to start is from the `vivarium research template <https://github.com/ihmeuw/vivarium_research_template>`_. In this repo's version of :code:`loader.py`
+you can find some basic examples of creating the loader functions.
+
+Most of the time, data is loaded using the :code:`load_standard_data` function,
+which pulls data from GBD into Vivarium formatting. However, there might be
+alternative functions for population data keys, or if you want data in non-standard
+formatting.
+
+Additionally, consider if you need all of the data to be saved. For
+example, in the child nutrition model, we don't use data for age
+groups over 5 years old, so we filter out some of the pulled data to
+limit the size of the artifact.
+
+For data keys that don't pull information from GBD, the process of writing
+or editing these functions is largely the same. Instead of :code:`load_standard_data`
+you would generally just use :code:`read_csv` from a specified data path. The
+path to each data input is stored in the :code:`paths.py` file. If you update a
+data input, update the reference in the :code:`paths.py` file to ensure you're
+pulling the most up to date information.
+
+Other Input Data
+----------------
+
+Data keys and their functions make up the bulk of the data pulling
+infrastructure for artifact building. However, there might be some other
+metadata that's important to be aware of. Most of this can be found
+in the :code:`metadata.py` file.
+
+This is information like draw counts, index columns, and locations that
+can be run. If you're starting a new model, you'll have to write
```

Comment:
I'm not sure what you mean by this? When you create a new model the `metadata.py` will get created automatically by the research template.